File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/panel/condition/Disjunction.java
package org.fbme.smvDebugger.panel.condition;

import java.util.Map;

public class Disjunction extends BinaryExpression {
  public Disjunction(final Expression first, final Expression second) {
    super(first, second);
  }

  @Override
  public boolean evaluate(final Map<String, String> itemSimpleNameToValueMap) {
    return first.evaluate(itemSimpleNameToValueMap) || second.evaluate(itemSimpleNameToValueMap);
  }

  @Override
  public String getSymbol() {
    return "||";
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/panel/condition/Disjunction.kt
package org.fbme.smvDebugger.panel.condition

class Disjunction(first: Expression, second: Expression) : BinaryExpression(first, second) {
    override fun evaluate(itemSimpleNameToValueMap: Map<String, String>): Boolean {
        return first.evaluate(itemSimpleNameToValueMap) || second.evaluate(itemSimpleNameToValueMap)
    }

    override val symbol: String
        get() = "||"
}
